src/runtime-rs/crates/hypervisor/src/qemu/qmp.rs

```diff
@@ -128,6 +142,153 @@ impl Qmp {
 
         Ok(hotunplugged)
     }
+
+    pub fn set_guest_memory_block_size(&mut self, size: u64) {
+        self.guest_memory_block_size = size;
+    }
+
+    pub fn guest_memory_block_size(&self) -> u64 {
+        self.guest_memory_block_size
+    }
+
+    pub fn hotplugged_memory_size(&mut self) -> Result<u64> {
+        let memory_frontends = self.qmp.execute(&qapi_qmp::query_memory_devices {})?;
+
+        let mut hotplugged_mem_size = 0_u64;
+
+        info!(sl!(), "hotplugged_memory_size(): iterating over dimms");
```

Comment:
The idea here (and even more generally) is to log the "what" is being done rather than "how" if possible.  While it's technically true that we're iterating over all frontends at the highest level, we only do that because it's not possible (or practical) to directly iterate over dimms exclusively, and so I'd consider that an implementation detail of the iteration.

Reply:
Good, and let's leave it as it is. 😉